activesupport/lib/active_support/actionable_error.rb

```diff
@@ -43,6 +135,27 @@ module ClassMethods
       def action(name, &block)
         _actions[name] = block
       end
+
+      # Trigger the current actionable error from a pre-existing if the
+      # condition, given as a block, returns a truthy value.
+      #
+      #   class SetupError < Error
+      #     include ActiveSupport::ActionableError
+      #
+      #     trigger on: ActiveRecord::RecordInvalid, if: -> error do
+      #       error.message.match?(InboundEmail.table_name)
+      #     end
+      #   end
+      #
+      # Note: For an actionable error to be triggered, its constant needs to be
+      # loaded. If it isn't autoloaded, the triggers won't be registered until
+      # it is. Keep that in mind, if the actionable error isn't triggered.
+      def trigger(options)
```

Comment:
```suggestion
      def trigger(on:, if:)
```

Gets the keyword errors for free.

Reply:
I cannot use `if` as a local variable though 🤷‍♂️

Reply:
That's what `binding.local_variable_get(:if)` in my other suggestion is for.

Reply:
Sure, I was trying to avoid the workarounds. Here it is, though.